### PR TITLE
Api integration

### PR DESCRIPTION
**Pull request by AlexFlFe**
#32 
Pull Request with the implementation of the API. 
The app calls for a determined method in each of the useTaskManager functions (addTasks, deleteTasks, etc). We also implemented a method that loads the tasks of the todoist app determined project so both the app and todoist have the same data an both update at the same time.

Unfortunately, due to the lack of time, we had to hardcorde the API Token and the Project ID inside the useTaskManager tasks.

### CODE REVIEW

**Review by AleXKiD31 — APPROVED** [review]
Se ve bien para mi.

**Review by JordiJT99 — APPROVED** [review]

**Review by Albamph — APPROVED** [review]
lgtm!

**Comment by kianMIC** [comment]
LGTM

**Review by kianMIC — APPROVED** [review]

**Comment by SaraVictorFernandez** [comment]
Don't forget to add that this PR is related to task #32

File: src/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+API_KEY=119851f333e3b56540f44a073814b5b44fe00f25
```
Comment (SaraVictorFernandez) [inline src/.env:1]: I'll allow it, but commiting your private API key to a public repository loses its purpose

File: src/hook/useTaskManager.tsx
Comment (SaraVictorFernandez) [inline src/hook/useTaskManager.tsx:1]: Having the logic centralized in this file makes this PR a lot easier to read 👍

File: src/hook/useTaskManager.tsx [hunk 2]
```diff
@@ -9,77 +13,161 @@ interface Column {
...
-
-  const addTask = (columnaId: string, taskName: string) => {
-    setColumns(columns.map(col =>
-      col.id === columnaId
-        ? { ...col, tasks: [...col.tasks, { id: uuidv4(), name: taskName }] }
-        : col
-    ));
+  const [columns, setColumns] = useState<Column[]>(() => {
+    const savedColumns = localStorage.getItem('columns');
+    return savedColumns ? JSON.parse(savedColumns) : [{ id: uuidv4(), name: 'Columna 1', tasks: [] }];
```
[the remark this answers is not in the document]
Comment (JordiJT99) [reply src/hook/useTaskManager.tsx:18]: Sabemos que el localstorage no es la manera más óptima de guardar y cargar información pero por falta de tiempo es la solución más rápida que hemos encontrado. En un futuro la idea es que la página muestre al iniciarse la información cargándola desde la API.

[the remark this answers is not in the document]
Comment (SaraVictorFernandez) [reply src/hook/useTaskManager.tsx:18]: 👍

**Review by SaraVictorFernandez — APPROVED** [review]

**Pull request merged**